src/osd/PeeringState.cc

```diff
@@ -5575,13 +5575,30 @@ boost::statechart::result PeeringState::Active::react(const AdvMap& advmap)
     ps->share_pg_info();
   }
 
+  bool need_acting_change = false;
   for (size_t i = 0; i < ps->want_acting.size(); i++) {
     int osd = ps->want_acting[i];
     if (!advmap.osdmap->is_up(osd)) {
       pg_shard_t osd_with_shard(osd, shard_id_t(i));
-      ceph_assert(ps->is_acting(osd_with_shard) || ps->is_up(osd_with_shard));
+      if (!ps->is_acting(osd_with_shard) && !ps->is_up(osd_with_shard)) {
+        need_acting_change = true;
```

Comment:
Can you add a debug statement here to print the osd id which is causing us to set need_acting_change to true? Since this was an assert earlier, we should be very clear that 023524a26d7e12e7ddfc3537582b1a1cb03af69e is the only reason we are hitting this.

Reply:
@neha-ojha Done.